src/main/java/software/amazon/documentdb/jdbc/DocumentDbConnection.java

```diff
@@ -100,46 +78,23 @@ public class DocumentDbConnection extends Connection
             }
             LOGGER.debug(sb.toString());
         }
-        this.session = createSshTunnel(connectionProperties);
+        maybeCreateSshTunnel(connectionProperties);
         initializeClients(connectionProperties);
     }
 
-    /**
-     * Initializes the SSH session and creates a port forwarding tunnel.
-     *
-     * @param connectionProperties the {@link DocumentDbConnectionProperties} connection properties.
-     * @return a {@link Session} session. This session must be closed by calling the
-     *          {@link Session#disconnect()} method.
-     * @throws SQLException if unable to create SSH session or create the port forwarding tunnel.
-     */
-    public static SshPortForwardingSession createSshTunnel(
-            final DocumentDbConnectionProperties connectionProperties) throws SQLException {
+    private void maybeCreateSshTunnel(final DocumentDbConnectionProperties connectionProperties) throws SQLException {
```

Comment:
Suggesting simplify the name for the function - `createSshTunnel`, and adding a javadoc explaining it creates a new SSH client or reuses an already existing one.

Reply:
@alexey-temnikov I can put the test for the need of SSH tunnel (`connectionProperties.enableSshTunnel()`) then use the name `ensureSshTunnel()`, instead?

Reply:
Excellent suggestion, Bruce!